monitoring-6.x.md

```diff
@@ -0,0 +1,948 @@
+
+```
+$ kubectl -nmonitoring get pods
+NAME                         READY   STATUS    RESTARTS   AGE
+alertmanager-main-0                    3/3     Running   0          27m
+alertmanager-main-1                    3/3     Running   0          26m
+alertmanager-main-2                    3/3     Running   0          26m
+grafana-6b645587d-chxxg                2/2     Running   0          27m
+kube-state-metrics-69594c468-wcr4g     3/3     Running   0          27m
+nethealth-4cjwh                        1/1     Running   0          26m
+node-exporter-hz972                    2/2     Running   0          27m
+prometheus-adapter-6586cf7b4f-hmwkf    1/1     Running   0          27m
+prometheus-k8s-0                       3/3     Running   1          26m
+prometheus-k8s-1                       0/3     Pending   0          26m
+prometheus-operator-7bd7d57788-mf8xn   1/1     Running   0          27m
+watcher-7b99cc55c-8qgms                1/1     Running   0          27m
+```
+
+Most of the cluster metrics are collected by Prometheus which uses the following in-cluster services:
+
+*   [node-exporter](https://github.com/prometheus/node_exporter) (collects hardware and OS metrics)
+*   [kube-state-metrics](https://github.com/kubernetes/kube-state-metrics) (collects Kubernetes resource metrics - deployments, nodes, pods)
+
+As mentioned, node-exporter collects hardware and OS metrics. There are various collectors that are dependent on each operating system. A list of the collectors can be found [here](https://github.com/prometheus/node_exporter#collectors). By default node_exporter will expose all metrics from enabled collectors. For advanced use the node_exporter can be passed an optional list of collectors to filter metrics. In Prometheus config you can use the following syntax:
+
+```
+ params:
+    collect[]:
+      - foo
+      - bar
+```
+
+Further, kube-state-metrics collects metrics about various Kubernetes resources such as deployments, nodes and pods. It is a service that listens to the Kubernetes API server and generates metrics about the state of the objects. 
+
+kube-state-metrics exposes raw data that is unmodified from the Kubernetes API, which allows users to have all the data they require and perform heuristics as they see fit. In return, kubectl may not show the same values, as kubectl applies certain heuristics to display cleaner messages.
+
+Metrics from kube-state-metrics service are exported on the HTTP endpoint `/metrics` on the listening port (default 8080) and are designed to be consumed by Prometheus.
+
+![diagram](https://miro.medium.com/max/832/1*7thrW4Wa5y6b03PxtPlQzA.jpeg) 
+
+(Source: https://medium.com/faun/production-grade-kubernetes-monitoring-using-prometheus-78144b835b60)
+
+All metrics collected by node-exporter and kube-state-metrics are placed into the k8s database in Prometheus. See below for a list of metrics collected by Prometheus. Each metric is stored as a separate “series” in Prometheus. 
```

Comment:
There is no concept of "database" in Prometheus as far as I'm aware. It just stores a bunch of time series. https://prometheus.io/docs/concepts/data_model/